test_pillow_pixel_data.py  AssertionError (76, 255, 28) == (255, 0, 0)

First encountered while building a debian package in a clean env and then reproduced on my laptop.
```
=================================== FAILURES ===================================
__________________________ test_PI_RGB[JPEG_RGB_RGB] ___________________________

test_with_pillow = [<module 'pydicom.pixel_data_handlers.numpy_handler' from '/home/yoh/deb/gits/p...build/cpython2_2.7_pydicom/build/pyd...handler' from '/home/yoh/deb/gits/pk...ybuild/cpython2_2.7_pydicom/build/pydicom/pixel_data_handlers/gdcm_handler.py'>]
image = '/home/yoh/deb/gits/pkg-exppsy/build-area/pydicom-1.1.0/.pybuild/cpython2_2.7_pydicom/build/pydicom/data/test_files/SC_rgb_dcmtk_+eb+cr.dcm'
PhotometricInterpretation = 'RGB'
results = [(255, 0, 0), (255, 128, 128), (0, 255, 0), (128, 255, 128), (0, 0, 255), (128, 128, 255), ...]
ground_truth = '/home/yoh/deb/gits/pkg-exppsy/build-area/pydicom-1.1.0/.pybuild/cpython2_2.7_pydicom/build/pydicom/data/test_files/SC_rgb_dcmtk_ebcr_dcmd.dcm'

    @pytest.mark.skipif(
        not test_pillow_jpeg_decoder,
        reason=pillow_missing_message)
    @pytest.mark.parametrize(
        "image,PhotometricInterpretation,results,ground_truth",
        testdata,
        ids=test_ids)
    def test_PI_RGB(test_with_pillow,
                    image,
                    PhotometricInterpretation,
                    results,
                    ground_truth):
        t = dcmread(image)
        assert t.PhotometricInterpretation == PhotometricInterpretation
        a = t.pixel_array
        assert a.shape == (100, 100, 3)
        """
        This complete test never gave a different result than
        just the 10 point test below
    
        gt = dcmread(ground_truth)
        b = gt.pixel_array
        for x in range(100):
            for y in range(100):
                assert tuple(a[x, y]) == tuple(b[x, y])
        """
        # this test points are from the ImageComments tag
>       assert tuple(a[5, 50, :]) == results[0]
E       assert (76, 255, 28) == (255, 0, 0)
E         At index 0 diff: 76 != 255
E         Use -v to get the full diff

pydicom/tests/test_pillow_pixel_data.py:591: AssertionError
=== 1 failed, 675 passed, 27 skipped, 5 xfailed, 2 xpassed in 10.30 seconds ====
E: pybuild pybuild:336: test: plugin distutils failed with: exit code=1: cd /home/yoh/deb/gits/pkg-exppsy/build-area/pydicom-1.1.0/.pybuild/cpython2_2.7_pydicom/build; python2.7 -m pytest 
```

any hints on what/where to dig would be appreciated!

#### Versions
<!--
Please run the following snippet and paste the output below.
import platform; print(platform.platform())
import sys; print("Python", sys.version)
import pydicom; print("pydicom", pydicom.__version__)
-->
```
Linux-4.15.0-2-amd64-x86_64-with-debian-buster-sid
('Python', '2.7.14+ (default, Mar 13 2018, 15:23:44) \n[GCC 7.3.0]')
('pydicom', '1.1.0')
```
PIL is 5.0.0-1 on the laptop and probably was 5.1.0 in the clean env (if relevant):

--- a/.travis.yml
+++ b/.travis.yml
@@ -46,7 +46,8 @@ matrix:
     - env: DISTRIB="conda" PYTHON_VERSION="3.7" NUMPY=true  PILLOW=both  JPEG2000=false JPEG_LS=true  GDCM=false
     - env: DISTRIB="conda" PYTHON_VERSION="3.7" NUMPY=true  PILLOW=both  JPEG2000=false JPEG_LS=true  GDCM=true
 
-    - env: DISTRIB="conda" PYTHON_VERSION="3.6" NUMPY=true  PILLOW=jpeg  JPEG2000=false JPEG_LS=false GDCM=false
+    - env: DISTRIB="conda" PYTHON_VERSION="2.7" NUMPY=true  PILLOW=jpeg  JPEG2000=false JPEG_LS=false GDCM=false
+    - env: DISTRIB="conda" PYTHON_VERSION="3.7" NUMPY=true  PILLOW=jpeg  JPEG2000=false JPEG_LS=false GDCM=false
 
     - env: DISTRIB="pypy" PYTHON_VERSION="2.7" NUMPY=false
     - env: DISTRIB="pypy" PYTHON_VERSION="2.7" NUMPY=true

--- a/.travis.yml
+++ b/.travis.yml
@@ -14,37 +14,39 @@ matrix:
     - env: DISTRIB="conda" PYTHON_VERSION="2.7" NUMPY=false PILLOW=false JPEG2000=false JPEG_LS=false GDCM=false
     - env: DISTRIB="conda" PYTHON_VERSION="2.7" NUMPY=true  PILLOW=false JPEG2000=false JPEG_LS=false GDCM=false
     - env: DISTRIB="conda" PYTHON_VERSION="2.7" NUMPY=true  PILLOW=false JPEG2000=false JPEG_LS=false GDCM=true
-    - env: DISTRIB="conda" PYTHON_VERSION="2.7" NUMPY=true  PILLOW=true  JPEG2000=false JPEG_LS=false GDCM=false
-    - env: DISTRIB="conda" PYTHON_VERSION="2.7" NUMPY=true  PILLOW=true  JPEG2000=false JPEG_LS=true  GDCM=false
-    - env: DISTRIB="conda" PYTHON_VERSION="2.7" NUMPY=true  PILLOW=true  JPEG2000=false JPEG_LS=true  GDCM=true
+    - env: DISTRIB="conda" PYTHON_VERSION="2.7" NUMPY=true  PILLOW=both  JPEG2000=false JPEG_LS=false GDCM=false
+    - env: DISTRIB="conda" PYTHON_VERSION="2.7" NUMPY=true  PILLOW=both  JPEG2000=false JPEG_LS=true  GDCM=false
+    - env: DISTRIB="conda" PYTHON_VERSION="2.7" NUMPY=true  PILLOW=both  JPEG2000=false JPEG_LS=true  GDCM=true
 
     - env: DISTRIB="conda" PYTHON_VERSION="3.4" NUMPY=false PILLOW=false JPEG2000=false JPEG_LS=false GDCM=false
     - env: DISTRIB="conda" PYTHON_VERSION="3.4" NUMPY=true  PILLOW=false JPEG2000=false JPEG_LS=false GDCM=false
     - env: DISTRIB="conda" PYTHON_VERSION="3.4" NUMPY=true  PILLOW=false JPEG2000=false JPEG_LS=false GDCM=true
-    - env: DISTRIB="conda" PYTHON_VERSION="3.4" NUMPY=true  PILLOW=true  JPEG2000=false JPEG_LS=false GDCM=false
-    - env: DISTRIB="conda" PYTHON_VERSION="3.4" NUMPY=true  PILLOW=true  JPEG2000=false JPEG_LS=true  GDCM=false
-    - env: DISTRIB="conda" PYTHON_VERSION="3.4" NUMPY=true  PILLOW=true  JPEG2000=false JPEG_LS=true  GDCM=true
+    - env: DISTRIB="conda" PYTHON_VERSION="3.4" NUMPY=true  PILLOW=both  JPEG2000=false JPEG_LS=false GDCM=false
+    - env: DISTRIB="conda" PYTHON_VERSION="3.4" NUMPY=true  PILLOW=both  JPEG2000=false JPEG_LS=true  GDCM=false
+    - env: DISTRIB="conda" PYTHON_VERSION="3.4" NUMPY=true  PILLOW=both  JPEG2000=false JPEG_LS=true  GDCM=true
 
     - env: DISTRIB="conda" PYTHON_VERSION="3.5" NUMPY=false PILLOW=false JPEG2000=false JPEG_LS=false GDCM=false
     - env: DISTRIB="conda" PYTHON_VERSION="3.5" NUMPY=true  PILLOW=false JPEG2000=false JPEG_LS=false GDCM=false
     - env: DISTRIB="conda" PYTHON_VERSION="3.5" NUMPY=true  PILLOW=false JPEG2000=false JPEG_LS=false GDCM=true
-    - env: DISTRIB="conda" PYTHON_VERSION="3.5" NUMPY=true  PILLOW=true  JPEG2000=false JPEG_LS=false GDCM=false
-    - env: DISTRIB="conda" PYTHON_VERSION="3.5" NUMPY=true  PILLOW=true  JPEG2000=false JPEG_LS=true  GDCM=false
-    - env: DISTRIB="conda" PYTHON_VERSION="3.5" NUMPY=true  PILLOW=true  JPEG2000=false JPEG_LS=true  GDCM=true
+    - env: DISTRIB="conda" PYTHON_VERSION="3.5" NUMPY=true  PILLOW=both  JPEG2000=false JPEG_LS=false GDCM=false
+    - env: DISTRIB="conda" PYTHON_VERSION="3.5" NUMPY=true  PILLOW=both  JPEG2000=false JPEG_LS=true  GDCM=false
+    - env: DISTRIB="conda" PYTHON_VERSION="3.5" NUMPY=true  PILLOW=both  JPEG2000=false JPEG_LS=true  GDCM=true
 
     - env: DISTRIB="conda" PYTHON_VERSION="3.6" NUMPY=false PILLOW=false JPEG2000=false JPEG_LS=false GDCM=false
     - env: DISTRIB="conda" PYTHON_VERSION="3.6" NUMPY=true  PILLOW=false JPEG2000=false JPEG_LS=false GDCM=false
     - env: DISTRIB="conda" PYTHON_VERSION="3.6" NUMPY=true  PILLOW=false JPEG2000=false JPEG_LS=false GDCM=true
-    - env: DISTRIB="conda" PYTHON_VERSION="3.6" NUMPY=true  PILLOW=true  JPEG2000=false JPEG_LS=false GDCM=false
-    - env: DISTRIB="conda" PYTHON_VERSION="3.6" NUMPY=true  PILLOW=true  JPEG2000=false JPEG_LS=true  GDCM=false
-    - env: DISTRIB="conda" PYTHON_VERSION="3.6" NUMPY=true  PILLOW=true  JPEG2000=false JPEG_LS=true  GDCM=true
+    - env: DISTRIB="conda" PYTHON_VERSION="3.6" NUMPY=true  PILLOW=both  JPEG2000=false JPEG_LS=false GDCM=false
+    - env: DISTRIB="conda" PYTHON_VERSION="3.6" NUMPY=true  PILLOW=both  JPEG2000=false JPEG_LS=true  GDCM=false
+    - env: DISTRIB="conda" PYTHON_VERSION="3.6" NUMPY=true  PILLOW=both  JPEG2000=false JPEG_LS=true  GDCM=true
 
     - env: DISTRIB="conda" PYTHON_VERSION="3.7" NUMPY=false PILLOW=false JPEG2000=false JPEG_LS=false GDCM=false
     - env: DISTRIB="conda" PYTHON_VERSION="3.7" NUMPY=true  PILLOW=false JPEG2000=false JPEG_LS=false GDCM=false
     - env: DISTRIB="conda" PYTHON_VERSION="3.7" NUMPY=true  PILLOW=false JPEG2000=false JPEG_LS=false GDCM=true
-    - env: DISTRIB="conda" PYTHON_VERSION="3.7" NUMPY=true  PILLOW=true  JPEG2000=false JPEG_LS=false GDCM=false
-    - env: DISTRIB="conda" PYTHON_VERSION="3.7" NUMPY=true  PILLOW=true  JPEG2000=false JPEG_LS=true  GDCM=false
-    - env: DISTRIB="conda" PYTHON_VERSION="3.7" NUMPY=true  PILLOW=true  JPEG2000=false JPEG_LS=true  GDCM=true
+    - env: DISTRIB="conda" PYTHON_VERSION="3.7" NUMPY=true  PILLOW=both  JPEG2000=false JPEG_LS=false GDCM=false
+    - env: DISTRIB="conda" PYTHON_VERSION="3.7" NUMPY=true  PILLOW=both  JPEG2000=false JPEG_LS=true  GDCM=false
+    - env: DISTRIB="conda" PYTHON_VERSION="3.7" NUMPY=true  PILLOW=both  JPEG2000=false JPEG_LS=true  GDCM=true
+
+    - env: DISTRIB="conda" PYTHON_VERSION="3.6" NUMPY=true  PILLOW=jpeg  JPEG2000=false JPEG_LS=false GDCM=false
 
     - env: DISTRIB="pypy" PYTHON_VERSION="2.7" NUMPY=false
     - env: DISTRIB="pypy" PYTHON_VERSION="2.7" NUMPY=true

--- a/build_tools/travis/install.sh
+++ b/build_tools/travis/install.sh
@@ -45,8 +45,11 @@ if [[ "$DISTRIB" == "conda" ]]; then
     if [[ "$JPEG2000" == "true" ]]; then
         sudo apt-get install libopenjp2-7 libopenjp2-7-dev
     fi
-    if [[ "$PILLOW" == "true" ]]; then
+    if [[ "$PILLOW" == "both" ]]; then
         conda install --yes pillow jpeg
+    elif [[ "$PILLOW" == "jpeg" ]]; then
+        conda install --yes jpeg
+        pip install pillow --global-option="build_ext" --global-option="--disable-jpeg2000"
     fi
     if [[ "$JPEG_LS" == "true" ]]; then
         conda install --yes cython

--- a/build_tools/travis/install.sh
+++ b/build_tools/travis/install.sh
@@ -47,9 +47,12 @@ if [[ "$DISTRIB" == "conda" ]]; then
     fi
     if [[ "$PILLOW" == "both" ]]; then
         conda install --yes pillow jpeg
+        python -c "from PIL import _imaging; print('JPEG plugin:', hasattr(_imaging, 'jpeg_decoder'))"
+        python -c "from PIL import _imaging; print('JPEG2k plugin:', hasattr(_imaging, 'jpeg2k_decoder'))"
     elif [[ "$PILLOW" == "jpeg" ]]; then
-        conda install --yes jpeg
         pip install pillow --global-option="build_ext" --global-option="--disable-jpeg2000"
+        python -c "from PIL import _imaging; print('JPEG plugin:', hasattr(_imaging, 'jpeg_decoder'))"
+        python -c "from PIL import _imaging; print('JPEG2k plugin:', hasattr(_imaging, 'jpeg2k_decoder'))"
     fi
     if [[ "$JPEG_LS" == "true" ]]; then
         conda install --yes cython

--- a/pydicom/pixel_data_handlers/pillow_handler.py
+++ b/pydicom/pixel_data_handlers/pillow_handler.py
@@ -160,10 +160,8 @@ def get_pixeldata(dicom_dataset):
         if dicom_dataset.BitsAllocated > 8:
             raise NotImplementedError("JPEG Lossy only supported if "
                                       "Bits Allocated = 8")
-        generic_jpeg_file_header = (
-            b'\xff\xd8\xff\xe0\x00\x10'
-            b'JFIF\x00\x01\x01\x01\x00\x01\x00\x01\x00\x00')
-        frame_start_from = 2
+        generic_jpeg_file_header = b''
+        frame_start_from = 0
     elif (dicom_dataset.file_meta.TransferSyntaxUID in
           PillowJPEG2000TransferSyntaxes):
         logger.debug("This is a JPEG 2000 format")
@@ -175,6 +173,7 @@ def get_pixeldata(dicom_dataset):
         logger.debug("This is a another pillow supported format")
         generic_jpeg_file_header = b''
         frame_start_from = 0
+
     try:
         UncompressedPixelData = bytearray()
         if ('NumberOfFrames' in dicom_dataset and

